docs/src/main/sphinx/security/certificate.rst

```diff
@@ -0,0 +1,108 @@
+    certificates for each client. Sites in this category have an existing PKI
+    infrastructure, possibly including an onsite Certificate Authority.
+
+    This feature is not appropriate for sites that would need to generate a set
+    of client certificates in order to use this authentication method. Consider
+    instead using another :doc:`authentication method
+    </security/authentication-types>`.
+
+Using certificate authentication
+--------------------------------
+
+All clients connecting with HTTPS/TLS go through the following initial steps:
+
+* The client attempts to contact the coordinator.
+* The coordinator returns its certificate to the client.
+* The client validates the server's certificate using the client's trust store.
+
+A cluster with certificate authentication enabled goes through the following
+additional steps:
+
+* The coordinator asks the client for its certificate.
+* The client responds with its certificate.
+* The coordinator verifies the client's certificate, using the coordinator's
+  trust store.
+
+Several rules emerge from these steps:
+
+1. Trust stores used by clients must include the certificate of the signer of
+   the coordinator's certificate.
+2. Trust stores used by coordinators must include the certificate of the signer
+   of client certificates.
+3. The trust stores used by the coordinator and clients do not need to be the
+   same.
+4. The certificate that verifies the coordinator does not need to be the same as
+   the certificate verifying clients.
+
+Trino validates based on the distinguished name (DN) from the X.509 ``Subject``
+field. The principal created for the connection is the common name (CN) from the
+``Subject`` field. You can use :doc:`user mapping </security/user-mapping>` to
+map the CN to a Trino user name.
+
+There are three levels of client certificate support possible. From the point of
+view of the server:
+
+* The server does not require a certificate from clients.
+* The server asks for a certificate from clients, but allows connection without one.
+* The server must have a certificate from clients to allow connection.
+
+Trino's client certificate support is the middle type. It asks for a certificate
+but allows connection if another authentication method passes.
+
+Certificate authentication configuration
+----------------------------------------
+
+Enable certificate authentication by setting the :doc:`Certificate
+authentication type <authentication-types>` in :ref:`etc/config.properties
+<config_properties>`:
+
+.. code-block:: properties
+
+    http-server.authentication.type=CERTIFICATE
+
+You can specify certificate authentication along with another authenticaton
+method, such as ``PASSWORD``. In this case, authentication is performed in the
+order of entries, and the first successful authentication results in access.
+
+.. code-block:: properties
+
+    http-server.authentication.type=CERTIFICATE,PASSWORD
```

Comment:
It might make sense to change the order to password first .. wdyt @dain ?

Reply:
CERT,PW leaves PW as a backup auth method for cases where a cert files to be presented, or fails validation. See above, that Trino asks for a cert, but allows connecting without one.

Reply:
I realise that .. but I think it might be more common to e.g. use cert only for a few automated systems and the majority of users are authenticating with LDAP (password) .. in that case it would imho make more sense to have PASSWORD first and fall through to cert.. but I dont know what is more common. My scenario or the reverse where the majority is cert auths. Or maybe in terms of performance is make basically no difference... hence I am asking @dain

Reply:
Makes sense. Maybe we should show several combinations and explain the scenarios where they might be used.

Reply:
I would add a `For example, the following...` sentence before this.

Reply:
done